service/src/test/kotlin/no/nav/su/se/bakover/service/søknadsbehandling/SøknadsbehandlingServiceIverksettTest.kt

```diff
@@ -190,104 +125,51 @@ internal class SøknadsbehandlingServiceIverksettTest {
-            behandlingMetrics = behandlingMetricsMock,
-            vedtakRepo = vedtakRepoMock,
-            kontrollsamtaleService = kontrollsamtaleServiceMock,
-            observer = statistikkObserver,
+            søknadsbehandlingRepo = mock {
+                on { hent(any()) } doReturn behandling
+            },
+            utbetalingService = mock {
+                on { genererUtbetalingsRequest(any(), any(), any(), any(), any()) } doReturn simulertUtbetaling.right()
+                on { publiserUtbetaling(any()) } doReturn utbetalingsRequest.right()
+            },
+            kontrollsamtaleService = mock {
+                on { opprettPlanlagtKontrollsamtale(any(), any()) } doReturn kontrollsamtale().right()
+            },
         )
+
         val response = serviceAndMocks.søknadsbehandlingService.iverksett(
             SøknadsbehandlingService.IverksettRequest(
                 behandlingId = behandling.id,
-                attestering = Attestering.Iverksatt(attestant, attesteringstidspunkt),
+                attestering = Attestering.Iverksatt(attestant, fixedTidspunkt),
             ),
         )
 
-        val expected = Søknadsbehandling.Iverksatt.Innvilget(
-            id = behandling.id,
-            opprettet = behandling.opprettet,
-            sakId = behandling.sakId,
-            saksnummer = behandling.saksnummer,
-            søknad = behandling.søknad,
-            oppgaveId = behandling.oppgaveId,
-            behandlingsinformasjon = behandling.behandlingsinformasjon,
-            fnr = behandling.fnr,
-            beregning = behandling.beregning,
-            simulering = behandling.simulering,
-            saksbehandler = behandling.saksbehandler,
-            attesteringer = Attesteringshistorikk.empty()
-                .leggTilNyAttestering(Attestering.Iverksatt(attestant, attesteringstidspunkt)),
-            fritekstTilBrev = "",
-            stønadsperiode = behandling.stønadsperiode,
-            grunnlagsdata = Grunnlagsdata.IkkeVurdert,
-            vilkårsvurderinger = Vilkårsvurderinger.Søknadsbehandling.IkkeVurdert,
-        )
+        val expected = søknadsbehandlingIverksattInnvilget().second.copy(id = behandling.id)
 
         response shouldBe expected.right()
 
-        verify(søknadsbehandlingRepoMock).hent(behandling.id)
-        verify(utbetalingServiceMock).genererUtbetalingsRequest(
-            sakId = argThat { it shouldBe sakId },
-            attestant = argThat { it shouldBe attestant },
-            beregning = argThat { it shouldBe TestBeregning },
-            simulering = argThat { it shouldBe simulering },
-            uføregrunnlag = argThat { it shouldBe emptyList() },
+        verify(serviceAndMocks.søknadsbehandlingRepo).hent(behandling.id)
+        verify(serviceAndMocks.utbetalingService).genererUtbetalingsRequest(any(), any(), any(), any(), any())
+        verify(serviceAndMocks.søknadsbehandlingRepo).lagre(eq(expected), anyOrNull())
```

Comment:
Dersom disse eq-sjekkene feiler, får vi ikke de diff-feilmeldingene som gjør livet litt enklere for de som debugger det. Derfor vi har laget den argThat-greia (kan jo hende vi kan lage den litt mer smooth og.

Reply:
Ja, okei. Aldri helt skjønt hvorfor eq plutselig har dukket opp noe steder, men hvis det er "deprecated" så er det jo et bra svar